core-api/src/main/java/com/optimizely/ab/config/parser/DefaultConfigParser.java

```diff
@@ -47,7 +65,22 @@ public static ConfigParser getInstance() {
     ConfigParser create() {
         ConfigParser configParser;
 
-        if (isPresent("com.fasterxml.jackson.databind.ObjectMapper")) {
+        String configParserName = PropertyUtils.get("default_parser");
+
+        if(configParserName != null && isPresent(configParserName)) {
+            if (ConfigParsers.JACKSON_CONFIG_PARSER.toString().equals(configParserName)) {
+                configParser = new JacksonConfigParser();
+            } else if (ConfigParsers.GSON_CONFIG_PARSER.toString().equals(configParserName)) {
+                configParser = new GsonConfigParser();
+            } else if (ConfigParsers.JSON_SIMPLE_CONFIG_PARSER.toString().equals(configParserName)) {
+                configParser = new JsonSimpleConfigParser();
+            } else if (ConfigParsers.JSON_CONFIG_PARSER.toString().equals(configParserName)) {
+                configParser = new JsonConfigParser();
+            } else {
+                throw new MissingJsonParserException("unable to locate a JSON parser. "
+                    + "Please see <link> for more information");
+            }
+        } else if (isPresent("com.fasterxml.jackson.databind.ObjectMapper")) {
```

Comment:
Now that we have the enum as defined above this changes from `if...else if...`, to:
```
        for (ConfigParserSupplier supplier: ConfigParserSupplier.valuesI()) {
            if (!supplier.isPresent()) {
                continue;
            }

            ConfigParser configParser = supplier.get();
            logger.info("using json parser: {}", configParser.getClass().getSimpleName());
            return configParser;
        }

        throw new MissingJsonParserException("unable to locate a JSON parser. "
            + "Please see <link> for more information");
```